ui/pages/confirmations/components/confirm/info/personal-sign/siwe-sign/siwe-sign.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { toHex } from '@metamask/controller-utils';
+
+import { NETWORK_TO_NAME_MAP } from '../../../../../../../../shared/constants/network';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../../../selectors';
+import { SignatureRequestType } from '../../../../../types/confirm';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowAddress,
+  ConfirmInfoRowText,
+} from '../../../../../../../components/app/confirm/info/row';
+import { formatDate } from '../../../../../utils/date';
+
+const SIWESignInfo: React.FC = () => {
+  const t = useI18nContext();
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as SignatureRequestType;
+
+  if (!currentConfirmation.msgParams?.siwe?.parsedMessage) {
+    return null;
+  }
+
+  const siweMessage = currentConfirmation.msgParams.siwe?.parsedMessage;
+  const chainId = toHex(siweMessage.chainId);
+
+  return (
+    <>
+      <ConfirmInfoRow label={t('siweOriginalMessage')}>
+        <ConfirmInfoRowText text={siweMessage.statement} />
+      </ConfirmInfoRow>
+      <ConfirmInfoRow label={t('siweURI')}>
+        <ConfirmInfoRowText text={siweMessage.domain} />
+      </ConfirmInfoRow>
+      <ConfirmInfoRow label={t('siweNetwork')}>
+        <ConfirmInfoRowText
+          text={
+            (NETWORK_TO_NAME_MAP as Record<string, string>)[chainId] ?? chainId
```

Comment:
Should we define this above as `network` to avoid the logic in the JSX?

Reply:
I did this change but IMHO some logic inside JSX is ok